### PR TITLE
Allow TBD files to be generated in non-wmo mode

### PR DESCRIPTION
**Pull request by huonw**
This goes with option 1 (using `-merge-modules`) from https://forums.swift.org/t/tbd-emission-without-whole-module-optimisation-merge-using-external-process-or-emit-during-merge-modules/14093

Fixes rdar://problem/33735706

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci please smoke test

**Comment by huonw** [comment]
@swift-ci please smoke test Linux

**Comment by huonw** [comment]
@swift-ci please smoke test

File: lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -2181,6 +2181,12 @@ static void addAuxiliaryOutput(Compilation &C, CommandOutput &output,
     // multiple primary inputs; really it should be associated with the build as
     // a whole -- derived OFM input "" -- but that's a more general thing to
     // fix.
+    if (requireArg.isValid() && !C.getArgs().getLastArg(requireArg)) {
```
Comment (davidungar) [inline lib/Driver/Driver.cpp]: This code looks out of place to me. How about folding this into the previous else on line 2173. In other words 2173-2188 would be:
````
else if (requreArg.isValid() ...) {
  ...
  return
} else {
  // Put the auxilliary file next to ....
````

File: lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -2752,22 +2765,18 @@ void Driver::chooseLoadedModuleTracePath(Compilation &C, const OutputInfo &OI,
...
-          /*TreatAsTopLevelOutput=*/true, workingDirectory, "tbd", Buf);
-
-      Output->setAdditionalOutputForType(file_types::TY_TBD, filename);
-    }
+  StringRef pathFromArgs;
+  if (const Arg *A = C.getArgs().getLastArg(options::OPT_emit_tbd_path)) {
+    pathFromArgs = A->getValue();
   }
+
+  addAuxiliaryOutput(C, *Output, file_types::TY_TBD, OI, OutputMap,
```
Comment (davidungar) [inline lib/Driver/Driver.cpp]: Nice factoring!

**Review by davidungar — COMMENTED** [review]
Sorry I didn't see this till now! LGTM, but I would like to see that one change I mentioned to clean up the control flow a bit.

**Comment by huonw** [comment]
@swift-ci please smoke test

**Pull request merged**